resources/js/Components/Table.vue

```diff
@@ -0,0 +1,229 @@
+<template>
+  <table
+    ref="Table"
+    :class="[
+      'wikit',
+      'wikit-Table',
+      `wikit-Table--linear-${breakpoint}`
+    ]"
+  >
+    <slot />
+  </table>
+</template>
+
+<script setup lang="ts">
+import { PropType, computed, ref } from 'vue';
+import { Breakpoint, validateBreakpoint } from '../types/Breakpoint';
+
+/**
+ * Tables display categorical information organized across rows and columns in
+ * order to facilitate the comparative analysis of data.
+ *
+ * The WiKit table component provides a wrapper around the common HTML table
+ * elements such as `<thead>`, `<tbody>`, `<tr>`, `<th>` and `<td>`, to apply
+ * design system styles to tabular data.
+ *
+ * Adding a `data-header` attribute to the cells allows us to maintain the column
+ * headers and display them in the table's linearized form to provide additional
+ * context.
+ *
+ * **Example:**
+ *
+ * ```html
+ * <td data-header="Column Header">Content Here</td>
+ * ```
+ */
+	const Table = ref('Table');
+	const props = defineProps({
+		linearize: {
+			type: String as PropType<Breakpoint>,
```

Comment:
I know there's a snazzy way to do this that involves not importing `PropType` to begin with. Please share it with me!